src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstanceEventsController.cs

```diff
@@ -0,0 +1,177 @@
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for. </param>
+        /// <returns>List of instance events.</returns>
+        /// GET  storage/api/v1/instanceevents/?instanceid={instanceId}
+        [HttpGet]
+        public async Task<ActionResult> GetAllInstanceEvents(string instanceId)
+        {
+            if (!string.IsNullOrEmpty(instanceId))
+            {
+                List<InstanceEvent> result = await _repository.ListAllInstanceEvents(instanceId);
+                if (result == null || result.Count == 0)
+                {
+                    return NotFound($"Did not find any instance events for instanceId={instanceId}");
+                }
+
+                return Ok(result);
+            }
+
+            return BadRequest("Unable to perform query.");
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id and listed event types from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for.</param>
+        /// <param name="eventTypes">List of event types to filter the events by./param>
+        /// <returns>List of intance events.</returns>
+        /// GET  storage/api/v1/instanceevents/GetByInstanceEventType?instanceId={instanceId}&eventTypes={eventType}&eventTypes={eventType}
+        [HttpGet]
+        [Route("GetByInstanceEventType")]
+        public async Task<ActionResult> GetInstanceEventsEventTypes(string instanceId, List<string> eventTypes)
+        {
+            // or array of types is empty
+            if (string.IsNullOrEmpty(instanceId) || eventTypes.Count == 0 || eventTypes == null)
+            {
+                return BadRequest("Unable to perform query.");
+            }
+
+            List<InstanceEvent> result = await _repository.ListInstanceEventsSpecificEventTypes(instanceId, eventTypes);
+
+            if (result == null || result.Count == 0)
+            {
+                return NotFound($"Did not find any instance events for instanceId={instanceId} with event type {string.Join(", ", eventTypes.ToArray())}");
+            }
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id and time frame from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for.</param>
+        /// <param name="from"> Lower bound for DateTime span to filter events by. Utc format and invariantCulture. </param>
+        /// <param name="to"> Upper bound for DateTime span to filter events by. Utc format and invariantCulture. </param>
+        /// <returns>List of instance events.</returns>
+        /// GET storage/api/v1/instanceevents/GetByTimeFrame?instanceId={instanceId}&from={fromtime}&to={totime}
+        [HttpGet]
+        [Route("GetByTimeFrame")]
```

Comment:
Should not have a specific route. Resource is: instance-events

Reply:
Updated the routes of all methods